### PR TITLE
refactor: approval using hub api

### PR DESCRIPTION
**Pull request by hanszhoou**
## Description
https://mask.atlassian.net/browse/MF-1506

Closes # (NO_ISSUE)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/web3-shared/base/src/specs/index.ts [hunk 1]
```diff
@@ -887,15 +893,17 @@ export interface Hub<ChainId, SchemaType, GasOption, Web3HubOptions = HubOptions
         initial?: Web3HubOptions,
     ) => Promise<Pageable<NonFungibleToken<ChainId, SchemaType>>>
     /** Get all approved fungible tokens of given account. */
-    getApprovedFungibleTokens?: (
+    getApprovedFungibleTokenSpenders?: (
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts:896]: Use `getApprovedFungibleTokens` and `getApprovedNonFungibleTokens` in hub.

Comment (hanszhoou) [inline packages/web3-shared/base/src/specs/index.ts:898]: I just renamed it.

File: packages/web3-shared/base/src/specs/index.ts [hunk 2]
```diff
@@ -386,18 +386,24 @@ export interface NonFungibleAsset<ChainId, SchemaType> extends NonFungibleToken<
  * Authorization about a fungible token.
  */
 export interface FungibleTokenAuthorization<ChainId, SchemaType> {
-    amount: string
-    recipient: string
-    token: FungibleToken<ChainId, SchemaType>
+    tokenInfo:  Pick<FungibleToken<ChainId, SchemaType>, 'address' | 'logoURL' | 'symbol' | 'name'>
+    address: string
+    name: string | undefined
+    logo: React.ReactNode | undefined
```
Comment (guanbinrui) [inline packages/web3-shared/base/src/specs/index.ts]:
```suggestion
    spender: {
        address: string
        name: string | undefined
        logo: React.ReactNode | undefined
    }
```

Comment (hanszhoou) [reply packages/web3-shared/base/src/specs/index.ts]: Rename `FungibleTokenAuthorization` to `FungibleTokenSpenderAuthorization`, and added code comments.

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**